lib/cylc/param_expand.py

```diff
@@ -72,7 +72,7 @@ def expand(template, params, results, values=None):
 # To extract all parameter lists e.g. 'm,n,o' (from '<m,n,o>').
 REC_P_GROUP = re.compile(r"<(.*?)>")
 # To extract parameter name and optional offset or value e.g. 'm-1'.
-REC_P_OFFS = re.compile(r'(\w+)(?:\s*([-+=]\s*[\w]+))?')
+REC_P_OFFS = re.compile(r'(\w+)([\-\+]\d+|=%s)?' % TaskID.NAME_SUFFIX_RE)
```

Comment:
We may need to permit arbitrary white-space between operators.

Reply:
White space should all be removed by now, see also #2534.

Reply:
Good. Could we sneak a little whitespace into one of the tasks in the test to make sure this doesn't get broken with #2534?

Reply:
OK. It is unfortunate that usage of this regular expression for parsing subsections in `[runtime]` does require white-space matching.